framework/src/wifi_manager/wifi_manager.c

```diff
@@ -17,129 +17,320 @@
+	} while (0)
+
+#define WIFIMGR_DEC_NUM_CLIENT								                                 \
+	do {													\
+		pthread_mutex_lock(&g_manager_info.info_lock);		\
+		g_manager_info.num_sta--;							\
+		pthread_mutex_unlock(&g_manager_info.info_lock);	\
+	} while (0)
+
+#define LOCK_WIFIMGR pthread_mutex_lock(&g_manager_info.state_lock)
+#define UNLOCK_WIFIMGR pthread_mutex_unlock(&g_manager_info.state_lock)
+
+//
+// Debugging
+//
+#define WM_LOG_START nvdbg("[WM] T%d\t%s:%d\n", getpid(), __FUNCTION__, __LINE__);
+#define WM_LOG_HANDLER_START nvdbg("[WM] T%d %s:%d state(%d) evt(%d)\n", getpid(), __FUNCTION__, __LINE__, g_manager_info.state, msg->event);
+
+/*
+ * global variables
+ */
+static _wifimgr_info_s g_manager_info = {{0}, {0}, 0, 0, 0,
+										 WIFIMGR_NOT_INIT, WIFIMGR_NOT_INIT, 0,
+										 PTHREAD_MUTEX_INITIALIZER,
+										 PTHREAD_MUTEX_INITIALIZER,
+										 {NULL, NULL, NULL, NULL, NULL} };
 #ifdef CONFIG_ENABLE_IOTIVITY
-mqd_t g_dw_nwevent_mqfd;
+static mqd_t g_dw_nwevent_mqfd;
 #endif
 
 /*
- * Iotivity stack receives network events from wifi manager module
- * this will be changed later
+ * internal functions
+ */
+//
+// wifi utils callbacks registered
+//
+static void _wifi_utils_connect_event(wifi_utils_result_e result, void *arg);
+static void _wifi_utils_disconnect_event(void *arg);
+static void _wifi_utils_join_event(void *arg); // it needs arg which contains sta infomation joined
+static void _wifi_utils_leave_event(void *arg); // it needs arg which contains a sta info left
+static void _wifi_utils_scan_done(wifi_utils_result_e result, wifi_utils_scan_list_s *slist, void *arg);
+static void _wifi_dhcpd_event(void); // join event from dhcpd
+
+//
+// internal APIs
+//
+static wifi_manager_result_e _convert_scan_info(wifi_manager_scan_info_s **wm_scan_list, wifi_utils_scan_list_s *wu_scan_list);
+static void _free_scan_info(wifi_manager_scan_info_s *wm_scan_list);
+static void _convert_state_to_info(connect_status_e *conn, wifi_manager_mode_e *mode, _wifimgr_state_e state);
+static wifi_manager_result_e _start_dhcpd(void);
+static wifi_manager_result_e _stop_dhcpd(void);
+static wifi_manager_result_e _get_ipaddr_dhcp(void);
+static wifi_manager_result_e _wifimgr_deinit(void);
+static wifi_manager_result_e _wifimgr_run_sta(void);
+static wifi_manager_result_e _wifimgr_connect_ap(wifi_manager_ap_config_s *config);
+static wifi_manager_result_e _wifimgr_disconnect_ap(void);
+static wifi_manager_result_e _wifimgr_run_softap(wifi_manager_softap_config_s *config);
+static wifi_manager_result_e _wifimgr_stop_softap(void);
+static wifi_manager_result_e _wifimgr_scan(void);
+
+//
+// user callback handler
+//
+static void _handle_user_cb(_wifimgr_usr_cb_type_e evt, void *arg);
+
+//
+// functions managing a state machine
+//
+static wifi_manager_result_e _handle_request(_wifimgr_msg_s *msg);
+static wifi_manager_result_e _handler_on_notinit_state(_wifimgr_msg_s *msg);
```

Comment:
I recommend to rename _handler_on_notinit_state with _handler_on_uninitialized_state.

Reply:
ok I will change